cluster/hosts.go

```diff
@@ -26,21 +27,19 @@ func (c *Cluster) TunnelHosts(ctx context.Context, local bool) error {
 		}
 		return nil
 	}
-	for i := range c.EtcdHosts {
-		if err := c.EtcdHosts[i].TunnelUp(ctx, c.DockerDialerFactory); err != nil {
-			return fmt.Errorf("Failed to set up SSH tunneling for Etcd host [%s]: %v", c.EtcdHosts[i].Address, err)
+	c.InactiveHosts = make([]*hosts.Host, 0)
+	uniqueHosts := hosts.GetUniqueHostList(c.EtcdHosts, c.ControlPlaneHosts, c.WorkerHosts)
+	for i := range uniqueHosts {
+		if err := uniqueHosts[i].TunnelUp(ctx, c.DockerDialerFactory); err != nil {
+			log.Warnf(ctx, "Failed to set up SSH tunneling for Control host [%s]: %v", uniqueHosts[i].Address, err)
+			c.InactiveHosts = append(c.InactiveHosts, uniqueHosts[i])
 		}
 	}
-	for i := range c.ControlPlaneHosts {
-		err := c.ControlPlaneHosts[i].TunnelUp(ctx, c.DockerDialerFactory)
-		if err != nil {
-			return fmt.Errorf("Failed to set up SSH tunneling for Control host [%s]: %v", c.ControlPlaneHosts[i].Address, err)
-		}
-	}
-	for i := range c.WorkerHosts {
-		if err := c.WorkerHosts[i].TunnelUp(ctx, c.DockerDialerFactory); err != nil {
-			return fmt.Errorf("Failed to set up SSH tunneling for Worker host [%s]: %v", c.WorkerHosts[i].Address, err)
-		}
+	for _, host := range c.InactiveHosts {
+		c.EtcdHosts = removeFromHosts(host, c.EtcdHosts)
```

Comment:
may be we should add the warning about host being skipped from the service node list, here as well.